microprofile/rsocket/server/src/main/java/io/helidon/microprofile/rsocket/server/RSocketCdiExtension.java

```diff
@@ -0,0 +1,271 @@
+import java.util.concurrent.Flow;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Initialized;
+import javax.enterprise.context.spi.CreationalContext;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AnnotatedMethod;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.DeploymentException;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.WithAnnotations;
+
+import io.helidon.common.reactive.Multi;
+import io.helidon.common.reactive.Single;
+import io.helidon.config.Config;
+import io.helidon.config.ConfigValue;
+import io.helidon.microprofile.cdi.RuntimeStart;
+import io.helidon.microprofile.server.ServerCdiExtension;
+import io.helidon.rsocket.server.RSocketEndpoint;
+import io.helidon.rsocket.server.RSocketRouting;
+import io.helidon.rsocket.server.RSocketSupport;
+import io.helidon.webserver.tyrus.TyrusSupport;
+
+import io.rsocket.Payload;
+
+
+
+import static javax.interceptor.Interceptor.Priority.PLATFORM_AFTER;
+
+/**
+ * Configure RSocket related things.
+ */
+public class RSocketCdiExtension implements Extension {
+    private static final Logger LOGGER = Logger.getLogger(RSocketCdiExtension.class.getName());
+
+    private Config config;
+
+    private ServerCdiExtension serverCdiExtension;
+
+    private Map<Class, Map<Annotation, Method>> methodMap = new HashMap<>();
+
+    private Map<String, RSocketRouting.Builder> routingMap = new HashMap<>();
+
+    private String rsocketPathRoot = "/rsocket";
+
+    /**
+     * Read Configuration.
+     *
+     * @param config Config
+     */
+    private void prepareRuntime(@Observes @RuntimeStart Config config) {
+        this.config = config;
+        ConfigValue<String> root = config.get("rsocket").get("root").asString();
+        if (root.isPresent()) {
+            rsocketPathRoot = root.get();
+        }
+    }
+
+    /**
+     * Register configured RSockets.
+     *
+     * @param event Event
+     * @param beanManager BeanManager
+     */
+    private void startServer(@Observes @Priority(PLATFORM_AFTER + 99) @Initialized(ApplicationScoped.class) Object event,
+                             BeanManager beanManager) {
+        serverCdiExtension = beanManager.getExtension(ServerCdiExtension.class);
+        wireMethods(beanManager);
+        registerRSockets();
+    }
+
+
+    /**
+     * Collect annotated endpoints.
+     *
+     * @param endpoint The endpoint.
+     */
+    private void endpointClasses(@Observes @WithAnnotations(RSocket.class) ProcessAnnotatedType<?> endpoint) {
+        LOGGER.info(() -> "Annotated endpoint found " + endpoint.getAnnotatedType().getJavaClass());
```

Comment:
Did you really intend this to be at "info" level?

Reply:
Fixed to "finest"